frontend/src/hooks/useProductDetail.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
```

Comment:
you should create a new file `/src/services/axiosBase` for the Axios call, if you set auth for your server, you should change only the base instead of all the files using it